crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,225 @@
+use axum::response::{IntoResponse, Response};
+
+use didcomm::{AttachmentData, Message};
+use hyper::StatusCode;
+use mongodb::bson::doc;
+use serde_json::{json, Value};
+
+use crate::{
+
+    model::stateful::entity::RoutedMessage, web::{error::MediationError, AppState, AppStateRepository}
+};
+
+/// mediator receives messages of type forward then it unpacks the messages and stores it for pickup
+/// the unpacked message is then repacked for further transmission.
+pub async fn mediator_forward_process(
+    state: &AppState,
+    payload: Message,
+) -> Result<Message, Response> {
+    let AppStateRepository {
+        message_repository,
+        connection_repository,
+        ..
+    } = state
+        .repository
+        .as_ref()
+        .ok_or_else(|| MediationError::RepostitoryError)
+        .unwrap();
+
+    // Check if the client's did in mediator's keylist
+
+    let next = payload.body.get("next").and_then(Value::as_str).unwrap();
```

Comment:
What if the **body** entry is malformed? unwrapping is not the right way to do in this case.